src/Microsoft.Data/BaseColumn.cs

```diff
@@ -110,6 +121,25 @@ protected set
         /// </param>
         protected internal virtual void AddDataViewColumn(DataViewSchema.Builder builder) => throw new NotImplementedException();
 
+        /// <summary>
+        /// Clips values beyond the specified thresholds
+        /// </summary>
+        /// <typeparam name="U"></typeparam>
+        /// <param name="lower">Minimum value. All values below this threshold will be set to it</param>
+        /// <param name="upper">Maximum value. All values above this threshold will be set to it</param>
+        public virtual BaseColumn Clip<U>(U lower, U upper) => throw new NotImplementedException();
+
+        /// <summary>
+        /// Determines if the column is of a numeric type
+        /// </summary>
+        /// <remarks>Used by the Description method to filter out descriptons from non-numeric columns</remarks>
+        public virtual bool IsNumericColumn() => false;
```

Comment:
I don't think this is the right design. Your `remarks` even suggests why naming this `IsNumericColumn` is a bad idea: `Used by the Description method to filter out descriptons from non-numeric columns` - why couldn't a non-numeric column want a description?

Instead, I would think this should be `HasDescription()` or `SupportsDescription()`. It could even be a property instead of a method.

Reply:
Sure, this is just the first pass at the `Description` method and I was planning on adding a minor improvement it in a follow up PR. The reason I went with `IsNumericColumn` is because `DataFrame.Description()` will call each `BaseColumn`'s `Description` method and then perform a `Merge`. The numeric columns have properties such as `Length, Max` etc, but the string columns don't have anything yet => The `Merge` will just produce an empty `DataFrame`, since there won't be any common properties. I could instead perform an `OuterMerge` which will preserve all the `Description`s, but I think we can inspect what columns are in the DataFrame and do something a little more intelligent here. I changed it to `HasDescription` for now.